ocrd_kraken/segment.py

```diff
@@ -1,62 +1,146 @@
-from __future__ import absolute_import
+from os.path import join
 
 from ocrd import Processor
-from ocrd_utils import getLogger, concat_padded, points_from_x0y0x1y1, MIMETYPE_PAGE
-from ocrd_models.ocrd_page import TextRegionType, TextLineType, CoordsType, to_xml
+from ocrd_utils import (
+    getLogger,
+    assert_file_grp_cardinality,
+    make_file_id,
+    concat_padded,
+    polygon_from_x0y0x1y1,
+    points_from_polygon,
+    coordinates_for_segment,
+    MIMETYPE_PAGE
+)
+from ocrd_models.ocrd_page import TextRegionType, TextLineType, CoordsType, BaselineType, to_xml
 from ocrd_modelfactory import page_from_file
 
-from ocrd_kraken.config import OCRD_TOOL
+import shapely.geometry as geom
+from shapely.prepared import prep as geom_prep
+from kraken.pageseg import segment as legacy_segment
 
-from kraken.pageseg import segment, detect_scripts
+from .config import OCRD_TOOL
 
 class KrakenSegment(Processor):
 
     def __init__(self, *args, **kwargs):
         kwargs['ocrd_tool'] = OCRD_TOOL['tools']['ocrd-kraken-segment']
+        kwargs['version'] = OCRD_TOOL['version']
         super(KrakenSegment, self).__init__(*args, **kwargs)
 
     def process(self):
         """
         Segment with kraken
         """
         log = getLogger('processor.KrakenSegment')
+        assert_file_grp_cardinality(self.input_file_grp, 1)
+        assert_file_grp_cardinality(self.output_file_grp, 1)
+        kwargs = {}
+        kwargs['text_direction'] = self.parameter['text_direction']
+        use_legacy = self.parameter['use_legacy']
+        if use_legacy:
+            kwargs['scale'] = self.parameter['scale']
+            kwargs['maxcolseps'] = self.parameter['maxcolseps']
+            kwargs['black_colseps'] = self.parameter['black_colseps']
+            log.info("Using legacy segmenter")
+            segment = legacy_segment
+        else:
+            from kraken.lib.vgsl import TorchVGSLModel
+            from kraken.blla import segment as blla_segment
+            log.info("Using blla segmenter")
+            blla_model_fname = self.resolve_resource(self.parameter['blla_model'])
+            kwargs['model'] = TorchVGSLModel.load_model(blla_model_fname)
```

Comment:
Note: we should move that whole chunk into a new `setup` function to be run outside of the `process` context – currently from the constructor (`if hasattr(self, 'output_file_grp')`).

Reply:
done